src/layman/upgrade/upgrade_v2_0.py

```diff
@@ -598,11 +598,77 @@ def migrate_layers():
         logger.warning(warning)
 
 
+def migrate_map_file_layer_names(layers_str: str, layer_name, layer_uuid, layer_ws) -> str:
+    layers = layers_str.split(',')
+    for layer_idx, full_layername in enumerate(layers):
+        layer_parts = full_layername.split(':')
+        if len(layer_parts) == 1:
+            if layer_parts[0] == layer_name:
+                layers[layer_idx] = f'l_{layer_uuid}'
+        elif len(layer_parts) == 2:
+            if layer_parts[0] in [f'{layer_ws}_wms', f'{layer_ws}'] and layer_parts[1] == layer_name:
+                wspace = 'layman_wms' if layer_parts[0] == f'{layer_ws}_wms' else 'layman'
+                layers[layer_idx] = f'{wspace}:l_{layer_uuid}'
+    return ','.join(layers)
+
+
+def migrate_map_file(map_file_path, map_layers):
+    with open(map_file_path, 'r', encoding="utf-8") as map_file:
+        map_json = json.load(map_file)
+    for map_layer in map_layers:
+        layer_index = map_layer['index']
+        layer_uuid = map_layer['uuid']
+        layer_ws = map_layer['workspace']
+        layer_name = map_layer['name']
+
+        layer_def = map_json['layers'][layer_index]
+        class_name = layer_def.get('className', '').split('.')[-1]
+        if class_name == 'WMS':
+            layer_def['url'] = layer_def['url'].replace(f'/geoserver/{layer_ws}_wms/', f'/geoserver/layman_wms/').replace(f'/geoserver/{layer_ws}/', f'/geoserver/layman/')
+            old_layers = layer_def['params']['LAYERS']
+            new_layers = migrate_map_file_layer_names(old_layers, layer_name, layer_uuid, layer_ws)
+            layer_def['params']['LAYERS'] = new_layers
+        elif class_name == 'Vector':
+            layer_def['protocol']['url'] = layer_def['protocol']['url'].replace(f'/{layer_ws}/', f'/layman/').replace(
+                f'/{layer_ws}_wms/', f'/layman_wms/')
+            old_layers = layer_def['name']
+            new_layers = migrate_map_file_layer_names(old_layers, layer_name, layer_uuid, layer_ws)
+            layer_def['name'] = new_layers
+        else:
+            raise ValueError(f'Unknown layer type: {layer_def["type"]}')
+
+    new_layers = get_layers_from_json(map_json)
+    old_layers = [(layer['uuid'], layer['index']) for layer in map_layers]
+    new_layers.sort(key=lambda x: (x[1], x[0]))
+    old_layers.sort(key=lambda x: (x[1], x[0]))
+    if new_layers != old_layers:
+        new_map_file_path = map_file_path + '.new'
+        with open(new_map_file_path, 'w', encoding="utf-8") as map_file:
+            json.dump(map_json, map_file, indent=4)
+        assert new_layers == old_layers, f'New internal layers do not correspond to the old ones. New file saved as {new_map_file_path}.\n{new_layers=}\n{old_layers=}'
```

Comment:
I suggest to make this assertion even if `map_layers` was initially empty list. Now the method `migrate_map_file` is called only if map_layers is not empty.

Reply:
Added.